eslint.config.base.js

```diff
@@ -0,0 +1,1266 @@
+    ...Object.keys(pkg.dependencies || {}),
+    ...Object.keys(pkg.devDependencies || {}),
+  ]);
+
+  return {
+    typescript: deps.has('typescript') && fs.existsSync(tsconfigPath),
+    react: deps.has('react'),
+    storybook: deps.has('storybook'),
+    vitest: deps.has('vitest'),
+    cypress: deps.has('cypress'),
+    reactTestingLibrary: deps.has('@testing-library/react'),
+    cypressTestingLibrary: deps.has('@testing-library/cypress'),
+  };
+}
+
+// eslint-disable-next-line complexity
+export function createConfig(opts = {}) {
+  const {
+    typescript: withTypeScript,
+    react: withReact,
+    storybook: withStorybook,
+    vitest: withVitest,
+    cypress: withCypress,
+    reactTestingLibrary: withReactTestingLibrary,
+    cypressTestingLibrary: withCypressTestingLibrary,
+  } = { ...DEFAULT_OPTS, ...opts };
+
+  return tseslint.config(
+    _.compact([
+      withTypeScript && {
+        /**
+         * Configure type-aware linting.
+         *
+         * To disable on a subset of files, see:
+         * https://typescript-eslint.io/getting-started/typed-linting#how-can-i-disable-type-aware-linting-for-a-subset-of-files
+         */
+        name: 'h5web/defaults/languages-ts',
+        languageOptions: {
+          parserOptions: {
+            projectService: true,
+            tsconfigRootDir: import.meta.dirname,
+          },
+        },
+      },
+      withReact && {
+        /**
+         * Enable JSX parsing.
+         */
+        name: 'h5web/defaults/languages-jsx',
+        files: ['**/*.{jsx,tsx}'],
+        languageOptions: {
+          parserOptions: {
+            ecmaFeatures: { jsx: true },
+            jsxPragma: null, // https://typescript-eslint.io/packages/parser/#jsxpragma
+          },
+        },
+      },
+      {
+        /**
+         * Configure browser globals on files we're confident aren't meant to run
+         * in a non-browser environment.
+         */
+        name: 'h5web/defaults/globals-browser',
+        files: ['**/*.{jsx,tsx}'],
+        languageOptions: {
+          globals: { ...globals.browser },
+        },
+      },
+      withVitest && {
+        /**
+         * Configure Vitest globals on test files.
+         */
+        name: 'h5web/defaults/globals-vitest',
+        files: [
+          '**/__tests__/**/*.{js,jsx,ts,tsx}',
+          '**/*.test.{js,jsx,ts,tsx}',
+        ],
+        languageOptions: {
+          globals: { ...vitestPlugin.environments.env.globals },
+        },
+      },
+      {
+        /**
+         * ESLint core rules: https://eslint.org/docs/latest/rules/
+         */
+        name: 'h5web/defaults/rules/core',
+        rules: {
+          'accessor-pairs': 'error',
+          'array-callback-return': [
+            'error',
+            { checkForEach: true }, // enforce no return in `forEach`
+          ],
+          // 'arrow-body-style': 'off', // use of curly braces in arrow functions should first be dictated by readabillity
```

Comment:
I declare every single rule explicitly so there are no surprises. I keep those that are disabled commented to not pollute the display when inspecting the config with `pnpm --filter @h5web/shared lint:eslint --inspect-config` or the like.